src/impl/android/external_commit_helper.cpp

```diff
@@ -89,24 +89,43 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
     int ret = mkfifo(path.c_str(), 0600);
     if (ret == -1) {
         int err = errno;
-        if (err == ENOTSUP) {
+        if (err == ENOTSUP || err == EACCES) {
             // Filesystem doesn't support named pipes, so try putting it in tmp instead
             // Hash collisions are okay here because they just result in doing
             // extra work, as opposed to correctness problems
-            std::ostringstream ss;
-
-            std::string tmp_dir(getenv("TMPDIR"));
-            ss << tmp_dir;
-            if (tmp_dir.back() != '/')
-              ss << '/';
-            ss << "realm_" << std::hash<std::string>()(path) << ".note";
-            path = ss.str();
-            ret = mkfifo(path.c_str(), 0600);
-            err = errno;
+
+            std::string tmp_dir = parent.get_temp_dir();
+            if (tmp_dir.empty()) {
+                auto tmp_dir_from_env = getenv("TMPDIR");
```

Comment:
If binding does not pass the temp directory and the `TMPDIR` environment variable also not set, this workaround is just disabled.

If we should throw an exception, I think it should be done at the initialization phase, not here.

Should the Object Store force to set the path of the temp dir?